devito/arch/compiler.py

```diff
@@ -601,7 +601,7 @@ def __init_finalize__(self, **kwargs):
         platform = kwargs.pop('platform', configuration['platform'])
 
         if platform is NVIDIAX:
-            self.cflags.append('-gpu=pinned')
+            self.cflags.append('-gpu=mem:separate:pinnedalloc')
```

Comment:
probably need a version check like icx for openmp

Reply:
(@georgebisbas ) pushed tweak so that older versions are guaranteed to work